sapi/fpm/fpm/fpm_request.c

```diff
@@ -243,7 +243,9 @@ void fpm_request_check_timed_out(struct fpm_child_s *child, struct timeval *now,
 	}
 #endif
 
-	if (proc.request_stage > FPM_REQUEST_ACCEPTING && ((proc.request_stage < FPM_REQUEST_END) || track_finished)) {
+	if (!fpm_request_is_idle(child) &&
+		proc.request_stage > FPM_REQUEST_ACCEPTING &&
```

Comment:
```suggestion
	if (proc.request_stage > FPM_REQUEST_READING_HEADERS &&
```

Reply:
the current version intentionally wraps that logic in `fpm_request_is_idle` - allowing other callers to use the same logic for detecting idle workers. What's wrong with that?